Readme Update - Workflow and Deployment

german-e-mas opened the pull request:
This PR updates the Readme with instructions on the Workflow and Deployment, along with some things to keep in mind.

ptal @basicNew

basicNew commented on README.md:
```diff
@@ -46,27 +47,55 @@ gem install bundler
+
+Make sure to checkout the `middleman` branch. This needs to be the starting point of your work branches.
+
+> The Ekumen Website is an [Organization GitHub Page](https://help.github.com/articles/user-organization-and-project-pages/), therefore, its static source files can be found in the `master` branch. Any push made into `master` will reflect into the website, so be careful!
```
I would put this right below the "Workflow" title

basicNew commented on README.md:
```diff
@@ -46,27 +47,55 @@ gem install bundler
+Once in your working branch, you can make changes to the website. Run the website locally to check that everything is working as expected.
+
+Make a Pull Request and change it's base to the `middleman` branch. After this has been properly reviewed and merged, the website needs to be manually built and deployed.
+
```
It is a bit confusing for me to have this section and then all of the (very small) next sections. Wouldn't it be better to have just one section detailing the steps with concrete examples? Like:

Make sure you are in the middleman branch:
```
$ git checkout middleman
```

Then run the website locally: 
```
$ middleman
```

etc, etc

basicNew commented on README.md:
```diff
@@ -46,27 +47,55 @@ gem install bundler
+
+This command will automatically push the `build` folder into the `master` branch, which will update the website. Note that the website won't load until the deploy is finished.
+
+The commit history you will see in the `master` branch depends on the `build` folder of each user, so don't worry about it. The important history lies in the `middleman` branch.
```
"on the `build` folder of each user" it is quite obscure; I would change that for something along the lines of "the `build` folder of your local repository" or "the `build` folder of your local installation"

basicNew commented on README.md:
```diff
@@ -46,27 +47,55 @@ gem install bundler
+
+## Rollback
+
+Something went wrong? Don't worry. You can always revert the `middleman` branch into a previous and stable state. Once this is done, you can build the website and deploy it again following the previously mentioned steps. This should be the correct workflow when it comes fixing problems.
```
"You can always revert the `middleman` branch into a previous and stable state." How? Again a couple of example command line elements can help a lot, esp if the person reading it just screwed things up.

basicNew commented on README.md:
```diff
@@ -46,27 +47,55 @@ gem install bundler
+
+Something went wrong? Don't worry. You can always revert the `middleman` branch into a previous and stable state. Once this is done, you can build the website and deploy it again following the previously mentioned steps. This should be the correct workflow when it comes fixing problems.
+
+In dire situations, you can commit and push directly into `master` and the website will reflect this change. Still, this is strongly discouraged and should be avoided.
```
- "dire" situations.
- Again, code example is a +100 here

german-e-mas commented:
I addressed the comments @basicNew ! I included more commands and removed some of the text, so it can be more straightforward.

ptal again when you come back :)

basicNew commented on README.md:
````diff
@@ -46,27 +47,66 @@ gem install bundler
+```
 
-You can deploy the website by making a Pull Request targetting the middleman branch. The website should be deployed automatically.
+Keep in mind that working directly on the `master` branch is discouraged and should be avoided. All the work done should have the `middleman` branch as a starting point.
````
Minor: add that, if you make changes in master and later re-deploy from middleman then the changes will be lost

basicNew commented:
Minor: "the Bundler" -> "Bundler"  (in the section title)

german-e-mas commented:
I addressed your comments @basicNew .
- Removed "the" when referring to `Bundler`.
- Stated that changes made in `master` will be lost in future deploys from `middleman`.

Can you ptal again?

german-e-mas commented:
Thanks for the review @basicNew ! Merging.
